packages/open-next/src/utils/stream.ts

```diff
@@ -0,0 +1,43 @@
+import { Readable } from "node:stream";
+import type { ReadableStream } from "node:stream/web";
+
+export function fromReadableStream(
+  stream: ReadableStream<Uint8Array>,
+  base64?: boolean,
+): Promise<string> {
+  const reader = stream.getReader();
+  const chunks: Uint8Array[] = [];
+
+  return new Promise((resolve, reject) => {
+    function pump() {
+      reader
+        .read()
+        .then(({ done, value }) => {
+          if (done) {
+            resolve(Buffer.concat(chunks).toString(base64 ? "base64" : "utf8"));
+            return;
+          }
+          chunks.push(value);
+          pump();
+        })
+        .catch(reject);
+    }
+    pump();
+  });
+}
+
+export function toReadableStream(
+  value: string,
+  isBase64?: boolean,
+): ReadableStream {
+  return Readable.toWeb(
+    Readable.from(Buffer.from(value, isBase64 ? "base64" : "utf8")),
+  );
+}
+
+export function emptyReadableStream(): ReadableStream {
+  if (process.env.OPEN_NEXT_FORCE_NON_EMPTY_RESPONSE === "true") {
+    return Readable.toWeb(Readable.from(["SOMETHING"]));
```

Comment:
Can this be an empty string or does it have to be a non-empty string?

Reply:
Empty string didn't work, we could have used `new Uint8Array(8)` but since it is used as a delimiter for the header part of the string i fear that this could introduce other issue later ( being misinterpreted by the streaming server )